### ISSUE TITLE
RyuJIT/x86: JitStress=2 CoreFx System.Runtime.Tests: Consistency check failed: AV in clr at this callstack: CORECLR! GetCLRRuntimeHost + 0x2A896 (0x630d35c6)

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#8232]
coreclr *(assuming root=c:\git\coreclr)*
```
build x86 checked skiptests
```
corefx *(assuming root=c:\git\corefx)*
```
build -Release -SkipTests -TestNugetRuntimeId=win7-x86 -- /p:BUILDTOOLS_OVERRIDE_RUNTIME=c:\git\coreclr\bin\Product\Windows_NT.x86.Checked
cd bin\tests\AnyOS.AnyCPU.Release\System.Runtime.Tests\default.netcoreapp1.1
set COMPlus_AltJit=*
set COMPlus_AltJitName=protojit.dll
set COMPlus_JitStress=2
RunTests.cmd c:\git\corefx\packages
```
**Output**
```
Using c:\git\corefx\packages as folder for resolving package dependencies.
Executing in C:\git\corefx\bin\tests\AnyOS.AnyCPU.Release\System.Runtime.Tests\default.netcoreapp1.1\
Hard linking dependent files...
Finished linking needed files, moving to running tests.
Running tests... Start time: 11:08:29.83
Command(s):
call CoreRun.exe xunit.console.netcore.exe System.Runtime.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapp1.1tests  -notrait category=OuterLoop -notrait category=failing -notrait category=nonwindowstests
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Runtime.Tests
Discovered:  System.Runtime.Tests
Starting:    System.Runtime.Tests
   System.Tests.UIntPtrTests.GetHashCodeRespectAllBits [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.Add [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.TestSize [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.Ctor_ULong [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.Subtract [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.TestCtor_VoidPointer_ToPointer [SKIP]
      Condition(s) not met: \"Is64Bit\"
   System.Tests.UIntPtrTests.TestImplicitCast [SKIP]
      Condition(s) not met: \"Is64Bit\"

Assert failure(PID 1316 [0x00000524], Thread: 7408 [0x1cf0]): Consistency check failed: AV in clr at this callstack:
------
CORECLR! GetCLRRuntimeHost + 0x2A896 (0x630d35c6)
<no module>! <no symbol> + 0x0 (0x02ea167d)
<no module>! <no symbol> + 0x0 (0x09d562d1)
<no module>! <no symbol> + 0x0 (0x09d55b21)
<no module>! <no symbol> + 0x0 (0x0bdaf838)
<no module>! <no symbol> + 0x0 (0x0bdae13e)
<no module>! <no symbol> + 0x0 (0x0bdaa3ae)
CORECLR! GetCLRRuntimeHost + 0x7B791 (0x631244c1)
CORECLR! GetCLRRuntimeHost + 0x2B24EA (0x6335b21a)
CORECLR! GetCLRRuntimeHost + 0x2B2686 (0x6335b3b6)
CORECLR! GetCLRRuntimeHost + 0x4F6406 (0x6359f136)
CORECLR! GetCLRRuntimeHost + 0x4EC2FE (0x6359502e)
CORECLR! GetCLRRuntimeHost + 0x4F44B3 (0x6359d1e3)
SYSTEM.PRIVATE.CORELIB.NI! <no symbol> + 0x0 (0x62b842a5)
SYSTEM.PRIVATE.CORELIB.NI! <no symbol> + 0x0 (0x62b84207)
<no module>! <no symbol> + 0x0 (0x09d53dd3)
<no module>! <no symbol> + 0x0 (0x09d53b9e)
<no module>! <no symbol> + 0x0 (0x09d5398a)
<no module>! <no symbol> + 0x0 (0x09d538fd)
-----
.AV on tid=0x1cf0 (7408), cxr=094FC228, exr=094FC1D8
FAILED: false

CORECLR! coreclr_shutdown + 0xB639 (0x6306a419)
CORECLR! GetCLRRuntimeHost + 0xDA5D2 (0x63183302)
CORECLR! GetCLRRuntimeHost + 0xDA27E (0x63182fae)
CORECLR! GetCLRRuntimeHost + 0xDA22B (0x63182f5b)
CORECLR! GetCLRRuntimeHost + 0xDA73B (0x6318346b)
NTDLL! LdrSetAppCompatDllRedirectionCallback + 0x1D6F3 (0x77ba0053)
NTDLL! RtlUnwind + 0x1BA (0x77b5e8fa)
NTDLL! KiUserExceptionDispatcher + 0xF (0x77b7002f)
<no module>! <no symbol> + 0x0 (0x02ea167d)
<no module>! <no symbol> + 0x0 (0x09d562d1)
    File: c:\git\corecl2\src\vm\excep.cpp Line: 8141
    Image: C:\git\corefx\bin\tests\AnyOS.AnyCPU.Release\System.Runtime.Tests\default.netcoreapp1.1\CoreRun.exe

Finished running tests.  End time=11:09:06.22, Exit code = 123456789
```

**Comment by pgavlin** [comment]
Investigating this is complicated somewhat by #8160, which frequently fires before the assertion mentioned in this issue.

**Comment by pgavlin** [comment]
CLosed by #8242.
